lib/ipfs/IpfsStorage.ts

```diff
@@ -44,10 +40,29 @@ export default class IpfsStorage {
     return IpfsStorage.ipfsStorageSingleton;
   }
 
-  private constructor (node: IPFS) {
+  private constructor (node: IPFS, repo: any) {
+    this.repo = repo;
     this.node = node;
   }
 
+  private static async getNode (repo?: any): Promise<IPFS> {
+    const localRepoName = 'sidetree-ipfs';
+    const options = {
+      repo: repo !== undefined ? repo : localRepoName
+    };
+    const node = await IPFS.create(options);
+    return node;
+  }
+
+  /**
+   * restarts the IPFS node
+   */
+  public async restart () {
```

Comment:
I am a bit uneasy that restart gets called by the handlers. Say 50 downloads/calls happens (which can happen, since these calls tend to come in batches within a block of transaction), then it seems `restart()` thus `node.stop()` can happen 50 times. An alternative design could be that we use a "monitor" that checks IPFS node state every 1 minute and only it restarts the node.

Reply:
Okay, I like this idea.